src/main/java/com/liferay/faces/util/product/ProductWrapper.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (c) 2000-2018 Liferay, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.liferay.faces.util.product;
+
+import javax.faces.FacesWrapper;
+
+
+/**
+ * @author  Kyle Stiemann
+ */
+public abstract class ProductWrapper implements Product, FacesWrapper<Product> {
```

Comment:
@stiemannkj1 If we determine that it is not possible to decorate the ProductFactory via faces-config.xml then ProductWrapper would be unnecessary.